signer/tests/integration/transaction_coordinator.rs

```diff
@@ -1169,26 +1155,45 @@ async fn run_dkg_if_signatures_required_changes(change_signatures_required: bool
         .await
         .expect("failed to write dkg shares");
 
-    let signer_set_info = SignerSetInfo {
-        aggregate_key: dkg_shares.aggregate_key,
-        // This matches the value we set for the bootstrap_signatures_required
-        signatures_required: ctx.config().signer.bootstrap_signatures_required,
-        signer_set: config_signer_set,
-    };
+    // Before any change DKG shouldn't be triggered
+    assert!(!should_coordinate_dkg(&ctx, &chaintip).await.unwrap());
+    assert!(assert_allow_dkg_begin(&ctx, &chaintip).await.is_err());
 
+    // Alter the registry based on the test
+    if scenario.signer_set_changed {
+        signer_set_info
+            .signer_set
+            .pop_first()
+            .expect("This signer set should not be empty");
+    }
+    if scenario.threshold_changed {
+        signer_set_info.signatures_required -= 1;
+    }
     ctx.state().update_registry_signer_set_info(signer_set_info);
 
-    // Create chaintip
-    let chaintip: model::BitcoinBlockRef = Faker.fake_with_rng(&mut rng);
-
-    // Before we actually change the signatures_required, the DKG won't be triggered
-    assert!(!should_coordinate_dkg(&ctx, &chaintip).await.unwrap());
-    assert!(assert_allow_dkg_begin(&ctx, &chaintip).await.is_err());
+    if scenario.latest_shares_matches {
+        // Latest shares must match the config
+        let mut signer_set: Vec<PublicKey> = ctx
+            .config()
+            .signer
+            .bootstrap_signing_set
+            .iter()
+            .copied()
+            .collect();
+        signer_set.sort();
```

Comment:
I don't think the sort is necessary since we are iterating over a `BTreeSet`, but I suppose it doesn't hurt.

Reply:
Oh right, removed in [357097a](https://github.com/stacks-sbtc/sbtc/pull/1795/commits/357097a45684e7eb30df19ca22596bc326e1f661)